src/optimization-settings.php

```diff
@@ -78,10 +78,25 @@ public function disable_frontend_scripts() {
 		public function load_frontend_scripts_conditionally( $block_content, $block ) {
 			if ( ! $this->is_script_loaded ) {
 				if ( get_option( 'stackable_optimize_script_load' ) && ! is_admin() ) {
-					if ( stripos( $block['blockName'], 'ugb/' ) === 0 ||
-						stripos( $block['blockName'], 'stackable/' ) === 0 ) {
+					if ( 
+						stripos( $block['blockName'], 'ugb/' ) === 0 ||
+						stripos( $block['blockName'], 'stackable/' ) === 0 
+					) {
 						stackable_block_assets();
 						$this->is_script_loaded = true;
+						return $block_content;
+					}
+
+					// When stackable blocks are displayed indirectly. Enqueue the assets.
+					preg_match( '/<!-- wp:ugb\//', $block_content, $ugb_matches );
+					preg_match( '/<!-- wp:stackable\//', $block_content, $stackable_matches );
```

Comment:
Please change these from regex searches into a faster string search since we're just looking for the presence of the string